aptos-move/aptos-vm/src/verifier/transaction_arg_validation.rs

```diff
@@ -344,12 +353,12 @@ pub(crate) fn recursively_construct_arg(
                 len -= 1;
             }
         },
-        Struct { idx, .. } | StructInstantiation { idx, .. } => {
-            let st = session
-                .fetch_struct_ty_by_idx(*idx, module_storage)
+        Struct { .. } | StructInstantiation { .. } => {
+            let (module_id, identifier) = session
+                .get_struct_name(ty, module_storage)
+                .map_err(|_| invalid_signature())?
```

Comment:
Is this correct - i.e. invalid_signature mapping?

Reply:
The code was using `.ok_or_else(invalid_signature)?;` before, so for compatibility I just re-used the mapping. The code is a mess and probably we need to rewrite it in any case at some point, without this kind of remapping, or at least with clear error messages.

Reply:
Btw, added a comment there specifying why we remap.